pkg/controller/statefulset/stateful_set.go

```diff
@@ -139,6 +139,11 @@ func NewStatefulSetController(
 				if oldPS.Status.Replicas != curPS.Status.Replicas {
 					logger.V(4).Info("Observed updated replica count for StatefulSet", "statefulSet", klog.KObj(curPS), "oldReplicas", oldPS.Status.Replicas, "newReplicas", curPS.Status.Replicas)
 				}
+				if curPS.ResourceVersion == oldPS.ResourceVersion {
```

Comment:
What is the benefit of introducing this kind of optimization? What are the potential risks?